File: TicTacToe/src/Spieler.java
public enum Spieler {
	LEER(" "), //
	KREUZ("X"), //
	KREIS("O");

	private final String muster;

	private Spieler(String muster) {
		this.muster = muster;
	}

	@Override
	public String toString() {
		return muster;
	}
}

File: TicTacToe/src/scalaCode/Spieler.scala
package scalaCode;

sealed trait Spieler {
  def istNiemand = false
  def istKreis = false
  def istKreuz = false
  def gegner : Spieler = Niemand
}

object Niemand extends Spieler {
  override def istNiemand = true
}
object Kreuz extends Spieler {
  override def istKreuz = true
  override def gegner = Kreis
}
object Kreis extends Spieler {
  override def istKreis = true
  override def gegner = Kreuz
}

